src/hotspot/share/runtime/signature.cpp

```diff
@@ -338,6 +338,11 @@ inline int SignatureStream::scan_type(BasicType type) {
 
   case T_ARRAY:
     while ((end < limit) && ((char)base[end] == JVM_SIGNATURE_ARRAY)) { end++; }
+    // If we discovered only the string of '[', this means something is wrong.
+    if (end >= limit) {
```

Comment:
Nit: Is > ever possible?

Reply:
Probably not, but for the sake of static analysis (bugs), I'd prefer to take no chances.

Reply:
@dholmes-ora, are you good with this explanation?

Reply:
Yes - thanks